androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/viewmodel/MapsViewModel.kt

```diff
@@ -0,0 +1,96 @@
+package com.kcc.kmmhackathon.androidHackathonApp.viewmodel
+
+import android.location.Location
+import androidx.annotation.RequiresPermission
+import androidx.hilt.lifecycle.ViewModelInject
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationCallback
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationResult
+import com.google.android.gms.maps.model.LatLng
+import com.kcc.kmmhackathon.shared.MealsSDK
+import com.kcc.kmmhackathon.shared.entity.Meal
+import com.kcc.kmmhackathon.shared.utility.DistanceUnit
+import kotlinx.coroutines.launch
+
+typealias Meals = List<Meal>
+
+class MapsViewModel @ViewModelInject constructor(
+    private val fusedLocationProviderClient: FusedLocationProviderClient
+) : ViewModel() {
+
+    sealed class State {
+        object LoadingMeals : State()
+        data class LoadedMeals(val meals: Meals, val distanceUnit: DistanceUnit) : State()
```

Comment:
Distance unit is currently unused but perhaps we could show the distance on the marker UI

Reply:
I think this is a good idea :)